templates/default/service-ndvm-pv

```diff
@@ -47,7 +47,7 @@
     "memory": "192",
     "display": "none",
     "cmdline": "root=\/dev\/xvda2 iommu=soft console=hvc0",
-    "kernel-extract": "0,1:\/bzImage",
+    "kernel": "\/usr\/lib\/xen\/boot\/grub-xen-pv64",
```

Comment:
You're going to need migration rules here to clear `kernel-extract`, and set `kernel`, for existing VMs on upgrade.  `upgrade-db/Migrations/M_37.hs` is a good reference.

Reply:
I had kinda thought I could get away without one.  The new service-ndvm and service-uivm would handle.  Existing ndvms and syncvms are already working and should keep working, so they don't need to change.

If the disk image were copied, then it wouldn't be updated with an appropriate grub.cfg.  So I guess a heuristic of "if type==ndvm and disk path==/storage/ndvm/ndvm.vhd, then update kernel config" would work.

Thanks for the review.

Reply:
Welcome.  If there are service VMs configs that don't utilize the "service-" template feature (that overwrites the db entries every boot), then they need a migration rule.